lib/src/repo.rs

```diff
@@ -1428,6 +1428,7 @@ impl MutableRepo {
             view.head_ids = self
                 .index()
                 .heads(&mut view.head_ids.iter())
+                .unwrap()
```

Comment:
nit: Add a comment here that we know that the commit should exist, as the default index impl guarantees it.

Reply:
I'm not sure the default implementation can't fail, but that still doesn't help non-default implementations. I think it's better to leave a TODO here (or update this function and its callers to propagate the error).

Reply:
The default implementation wouldn't fail (because all input commits must be valid), but I don't know if that applies to the Google backend. It's probably better to use different error type to clarify what kind of errors are expected.

I don't think it would make sense to propagate error from this function, at leas with the current design. `enforce_view_invariants()` exists for lazy computation, so it can be called from arbitrary places where the method call is supposed to be immutable & infallible.

Reply:
The Google backend does RPCs so it could fail. I think @AM5800 wanted to avoid crashing server jobs if that happens (even the server jobs do RPCs to the database). Maybe it's unlikely enough that we don't have to worry about it for now.

Reply:
After this conversation, I think also adding a TODO comment is approriate to maybe redesign the model to allow intermediate index failures and the comment should mention that we assume it for _all_ index implementations. 

> The default implementation wouldn't fail (because all input commits must be valid), but I don't know if that applies to the Google backend

That's what I assumed, so thank you for confirming it.

Reply:
Added a TODO